l10n_it_declaration_of_intent/models/account_move.py

```diff
@@ -278,6 +301,12 @@ def check_declarations_amounts(self, declarations):
 
     def get_declaration_residual_amounts(self, declarations):
         """Get residual amount for every `declarations`."""
+        available_plafond = 0.0
+        if self.move_type in ["in_invoice", "in_refund"]:
+            plafond = self.company_id.declaration_yearly_limit_ids.filtered(
+                lambda r: r.year == str(fields.first(declarations).date_start.year)
```

Comment:
Mi pare di capire che qui si dia per scontato che non possono esserci dichiarazioni di anni diversi.
Tipo una fattura di inizio anno non può scaricare dichiarazioni di due anni diversi?

Reply:
Ehm, se noti è un commit di semplice rifinitura, in cui ho limitato la funzione alle fatture di acquisto.

Non so comunque la logica che ci sta sotto perché sia così, concordo che è meglio informarsi con i funzionali.